### PR TITLE
fix: Edited changes are saved on orientation change.

### LINKED ISSUE #2702: Content of stickers removed when changing the orientation
**Describe the bug**
Stickers removed from image after changing the orientation.

**To Reproduce**

1. Select image and click on edit image 
2. Go to stickers section then select one sticker and apply it.
3. Change the orientation of the phone

**Expected behavior**
Sticker should remain on the image, it should not disappear.

**Smartphone Info:**

|               | |
|---------------|-|
|Device         |Redmi 5A|
|Android Version|7  |

**Would you like to work on the issue?**

- Yes

### LINKED ISSUE #2789: Edits in image not saved on orientation change.
**Describe the bug**
While editing the image if orientation is changed, the changes made in the image are not saved.

**Expected behavior**
The changes made should be saved even on orientation change.

**Would you like to work on the issue?**

- [X] Yes
- [ ] No
- Other:

### PR DESCRIPTION
**Pull request by yashk2000**
Fixed #2702 
Fixed #2789 

Changes: All edits in an image are stored when orientation change happens.

GIF:

ezgif com-video-to-gif (2)

### CODE REVIEW

**Comment by yashk2000** [comment]
@iamareebjamal @abishekvashok @mariobehling please review.

**Comment by yashk2000** [comment]
@adityastic please review.

**Comment by mariobehling** [comment]
Please provide a gif that is slower and a gif that does not use pictures of another app as example. Phimpe me is a photo app mainly. Could you please use photos to avoid confusion of users by using screenshots of apps to showcase the app?

**Comment by yashk2000** [comment]
@mariobehling sure. Just a minute.

**Comment by yashk2000** [comment]
@mariobehling gif updated.

File: app/src/main/java/org/fossasia/phimpme/editor/fragment/CropFragment.java [hunk 1]
```diff
@@ -39,11 +39,13 @@
   private LinearLayout ratioList, imageList, combinedList;
   private static List<RatioItem> dataList = new ArrayList<RatioItem>();
   private List<TextView> textViewList = new ArrayList<TextView>();
+  private static RectF rec;
 
   public static int SELECTED_COLOR = Color.BLUE;
   public static int UNSELECTED_COLOR = Color.BLACK;
   private CropRationClick mCropRationClick = new CropRationClick();
   public TextView selctedTextView;
+  private static TextView sView;
```
Comment (adityastic) [inline app/src/main/java/org/fossasia/phimpme/editor/fragment/CropFragment.java]: Dont. This is a memory leak

File: app/src/main/java/org/fossasia/phimpme/editor/EditImageActivity.java [hunk 2]
```diff
@@ -90,7 +90,7 @@
   private int initalBottomFragment = 0;
   private int initalMiddleFragment = 2;
   /** Number of times image has been edited. Indicates whether image has been edited or not. */
-  protected int mOpTimes = 0;
+  protected static int mOpTimes = 0;
```
Comment (adityastic) [inline app/src/main/java/org/fossasia/phimpme/editor/EditImageActivity.java]: dont make anything static, save this also in the savedState

File: app/src/main/java/org/fossasia/phimpme/editor/fragment/AddTextFragment.java [hunk 3]
```diff
@@ -45,6 +45,7 @@
 
   private InputMethodManager imm;
   private SaveTextStickerTask mSaveTask;
+  private static String text = null;
```
Comment (adityastic) [inline app/src/main/java/org/fossasia/phimpme/editor/fragment/AddTextFragment.java]: same, static variable will not be garbage collected

File: app/src/main/java/org/fossasia/phimpme/editor/fragment/CropFragment.java [hunk 4]
```diff
@@ -39,11 +39,13 @@
   private LinearLayout ratioList, imageList, combinedList;
   private static List<RatioItem> dataList = new ArrayList<RatioItem>();
   private List<TextView> textViewList = new ArrayList<TextView>();
+  private static RectF rec;
```
Comment (adityastic) [inline app/src/main/java/org/fossasia/phimpme/editor/fragment/CropFragment.java]: same

File: app/src/main/java/org/fossasia/phimpme/editor/fragment/CropFragment.java [hunk 5]
```diff
@@ -39,11 +39,13 @@
   private LinearLayout ratioList, imageList, combinedList;
   private static List<RatioItem> dataList = new ArrayList<RatioItem>();
```
Comment (adityastic) [inline app/src/main/java/org/fossasia/phimpme/editor/fragment/CropFragment.java]: even here xD

File: app/src/main/java/org/fossasia/phimpme/editor/fragment/RecyclerMenuFragment.java [hunk 6]
```diff
@@ -28,7 +28,7 @@
 
 public class RecyclerMenuFragment extends BaseEditFragment {
 
-  RecyclerView recyclerView;
+  private static RecyclerView recyclerView;
```
Comment (adityastic) [inline app/src/main/java/org/fossasia/phimpme/editor/fragment/RecyclerMenuFragment.java]: xD

File: app/src/main/java/org/fossasia/phimpme/editor/fragment/RotateFragment.java [hunk 7]
```diff
@@ -30,6 +30,7 @@
 
   private HorizontalWheelView horizontalWheelView;
   private TextView tvAngle;
+  private static double angle = Integer.MIN_VALUE;
```
Comment (adityastic) [inline app/src/main/java/org/fossasia/phimpme/editor/fragment/RotateFragment.java]: same, save it in saveInstance

File: app/src/main/java/org/fossasia/phimpme/editor/fragment/SliderFragment.java [hunk 8]
```diff
@@ -22,6 +25,7 @@
   SeekBar seekBar;
   ImageButton cancel, apply;
   public Bitmap filterBit;
+  private static int SeekBarProgress;
```
Comment (adityastic) [inline app/src/main/java/org/fossasia/phimpme/editor/fragment/SliderFragment.java]: same

File: app/src/main/java/org/fossasia/phimpme/editor/view/CustomPaintView.java [hunk 9]
```diff
@@ -18,10 +18,10 @@
 /** Created by panyi on 17/2/11. */
 public class CustomPaintView extends View {
   private Paint mPaint;
-  private Bitmap mDrawBit;
+  public Bitmap mDrawBit;
   private Paint mEraserPaint;
 
-  private Canvas mPaintCanvas = null;
+  public static Canvas mPaintCanvas = null;
```
Comment (adityastic) [inline app/src/main/java/org/fossasia/phimpme/editor/view/CustomPaintView.java]: .

File: app/src/main/java/org/fossasia/phimpme/editor/view/RotateImageView.java [hunk 10]
```diff
@@ -13,7 +13,7 @@
 
 public class RotateImageView extends View {
   private Rect srcRect;
-  private RectF dstRect;
+  private static RectF dstRect;
```
Comment (adityastic) [inline app/src/main/java/org/fossasia/phimpme/editor/view/RotateImageView.java]: .

File: app/src/main/java/org/fossasia/phimpme/editor/view/RotateImageView.java [hunk 11]
```diff
@@ -25,6 +25,7 @@
   private RectF wrapRect = new RectF(); // Picture surrounded by rectangles
   private Paint bottomPaint;
   private RectF originImageRect;
+  private static RectF img;
```
Comment (adityastic) [inline app/src/main/java/org/fossasia/phimpme/editor/view/RotateImageView.java]: .

File: app/src/main/java/org/fossasia/phimpme/editor/view/StickerView.java [hunk 12]
```diff
@@ -35,8 +35,7 @@
   public ImageViewTouch mainImage;
   private float leftX, rightX, topY, bottomY;
 
-  private LinkedHashMap<Integer, StickerItem> bank =
-      new LinkedHashMap<Integer, StickerItem>(); // Storing each data map
+  public static LinkedHashMap<Integer, StickerItem> bank;
```
Comment (adityastic) [inline app/src/main/java/org/fossasia/phimpme/editor/view/StickerView.java]: .

File: app/src/main/java/org/fossasia/phimpme/editor/fragment/CropFragment.java (hunk 5 above)
Comment (yashk2000) [reply app/src/main/java/org/fossasia/phimpme/editor/fragment/CropFragment.java]: I'll leave this one. This one wasn't done by me. It's been there from the beginning.

Comment (adityastic) [reply app/src/main/java/org/fossasia/phimpme/editor/fragment/CropFragment.java]: Ok, No worries

File: app/src/main/java/org/fossasia/phimpme/editor/view/CustomPaintView.java (hunk 9 above)
Comment (yashk2000) [reply app/src/main/java/org/fossasia/phimpme/editor/view/CustomPaintView.java]: Cannot avoid this one. I need to use the variable in other classes to retain state.

Comment (adityastic) [reply app/src/main/java/org/fossasia/phimpme/editor/view/CustomPaintView.java]: you can retain the view in this class itself

**Comment by yashk2000** [comment]
@adityastic done. I've only kept static in places where it was really needed. Rest have been removed.

File: app/src/main/java/org/fossasia/phimpme/editor/view/CustomPaintView.java (hunk 9 above)
Comment (yashk2000) [reply app/src/main/java/org/fossasia/phimpme/editor/view/CustomPaintView.java]: Tried that. The app crashed. When I reverted, it worked.

**Comment by yashk2000** [comment]
@abishekvashok @iamareebjamal @mariobehling please review and merge.

**Comment by yashk2000** [comment]
@mariobehling @abishekvashok can this please be merged? We can then proceed with a new release of the app.

**Comment by yashk2000** [comment]
@abishekvashok @iamareebjamal @mariobehling please merge.

**Comment by abishekvashok** [comment]
Thanks! Great job!

**Pull request merged**